src/LinkDotNet.Blog.Web/Features/ShowBlogPost/ShowBlogPostPage.razor

```diff
@@ -129,6 +134,10 @@ else if (BlogPost is not null)
     {
         await JsRuntime.InvokeVoidAsync("hljs.highlightAll");
         _ = UserRecordService.StoreUserRecordAsync();
+
+        ArgumentNullException.ThrowIfNull(BlogPost);
```

Comment:
As this isn't an argument, the `ArgumentNullException` doesn't really apply.
Rather an `if(BlogPost is not null)` - if the user navigates to a page with a non-existing ID then we get an exception